Author: Bo YU <tsu.yubo@gmail.com>, Sergei Golovan <sgolovan@debian.org>
Description: Patch increases timeouts in elixir tests.
Last-Modified: Fri, 31 Jan 2025 16:15:17 +0300
Bug-Debian: https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=1093147

diff --git a/lib/elixir/test/elixir/base_test.exs b/lib/elixir/test/elixir/base_test.exs
index aeb5f4b5c..d55d49234 100644
--- a/lib/elixir/test/elixir/base_test.exs
+++ b/lib/elixir/test/elixir/base_test.exs
@@ -1055,6 +1055,7 @@ test "hex_valid32?/2 with :mixed case and ignoring padding" do
   end
 
   # TODO: add valid? tests
+  @tag timeout: 120_000
   test "encode then decode is identity" do
     for {encode, decode, valid?} <- [
           {&encode16/2, &decode16!/2, &valid16?/2},
diff --git a/lib/elixir/test/elixir/kernel_test.exs b/lib/elixir/test/elixir/kernel_test.exs
index b4f1033fc..d229bade3 100644
--- a/lib/elixir/test/elixir/kernel_test.exs
+++ b/lib/elixir/test/elixir/kernel_test.exs
@@ -483,7 +483,7 @@ test "returns 0..-1//1" do
 
   describe "in/2" do
     # This test may take a long time on machine with low resources
-    @tag timeout: 120_000
+    @tag timeout: 240_000
     test "too large list in guards" do
       defmodule TooLargeList do
         @list Enum.map(1..1024, & &1)
diff --git a/lib/ex_unit/mix.exs b/lib/ex_unit/mix.exs
index 7f5e7fa12..ddc29c872 100644
--- a/lib/ex_unit/mix.exs
+++ b/lib/ex_unit/mix.exs
@@ -22,7 +22,7 @@ def application do
         # max_cases: System.schedulers_online * 2,
         # seed: rand(),
 
-        assert_receive_timeout: 100,
+        assert_receive_timeout: 600,
         autorun: true,
         capture_log: false,
         colors: [],
diff --git a/lib/mix/test/mix/tasks/compile_test.exs b/lib/mix/test/mix/tasks/compile_test.exs
index d20210279..25e27fd3c 100644
--- a/lib/mix/test/mix/tasks/compile_test.exs
+++ b/lib/mix/test/mix/tasks/compile_test.exs
@@ -390,7 +390,7 @@ test "code path pruning disabled" do
   end
 
   test "listening to concurrent compilations" do
-    timeout = 2_000
+    timeout = 8_000
 
     Mix.Project.pop()
 
diff --git a/lib/mix/test/mix/tasks/release_test.exs b/lib/mix/test/mix/tasks/release_test.exs
index 414802821..d04cb67b0 100644
--- a/lib/mix/test/mix/tasks/release_test.exs
+++ b/lib/mix/test/mix/tasks/release_test.exs
@@ -717,6 +717,7 @@ test "runs eval and version commands" do
   end
 
   @tag :unix
+  @tag timeout: 120_000
   test "runs in daemon mode" do
     in_fixture("release_test", fn ->
       config = [releases: [permanent2: [include_erts: false, cookie: "abcdefghij"]]]
diff --git a/lib/mix/test/mix/tasks/test_test.exs b/lib/mix/test/mix/tasks/test_test.exs
index 3c92a5a2e..29501b2ff 100644
--- a/lib/mix/test/mix/tasks/test_test.exs
+++ b/lib/mix/test/mix/tasks/test_test.exs
@@ -756,9 +756,9 @@ defp receive_until_match(port, expected, acc) do
           receive_until_match(port, expected, acc)
         end
     after
-      15_000 ->
+      45_000 ->
         raise """
-        nothing received from port after 15s.
+        nothing received from port after 45s.
         Expected: #{inspect(expected)}
         Got: #{inspect(acc)}
         """
